chapter-4.qmd

```diff
@@ -0,0 +1,267 @@
+---
+title: "Data transformation"
+---
+
+
+
+## Questions
+
+
+
+
+```{r}
+library(nycflights13)
+library(tidyverse)
+#> ── Attaching packages ──────────────────────────────────── tidyverse 1.3.2 ──
+#> ✔ ggplot2 3.4.0           ✔ purrr   1.0.1.9000 
+#> ✔ tibble  3.1.8           ✔ dplyr   1.0.99.9000
+#> ✔ tidyr   1.2.1.9001      ✔ stringr 1.5.0      
+#> ✔ readr   2.1.3           ✔ forcats 0.5.2      
+#> ── Conflicts ─────────────────────────────────────── tidyverse_conflicts() ──
+#> ✖ dplyr::filter() masks stats::filter()
+#> ✖ dplyr::lag()    masks stats::lag()
+```
+
+##Solutions
+
+## 4.2.5 Exercises
+
+1. Find all flights that
+
+a. Had a arrival delay of two hours or more
+
+```{r}
+
+
+filter(flights,arr_delay >= 2) %>% arrange(desc(arr_delay))
+```
+
+b. Flew to houston (IAH or HOU)
+
+```{r}
+
+glimpse(flights)
+view(flights)
+```
+
+```{r}
+
+flights %>% filter(dest %in% c("IAH","HOU"))
+```
+
+c. Were operated by United, American, or Delta
+
+```{r}
+flights %>% filter(carrier %in% c("UA","DL","AA"))
```

Comment:
Apply consistent code style, I recommend reviewing https://style.tidyverse.org/ and using the styler package's "Style active document" addin: https://styler.r-lib.org/.